### ISSUE TITLE
[ENH] Add real world datasets into dataset module

### ISSUE THREAD

**Issue by hazrulakmal** [sktime/sktime#4695]
To conduct a real-world benchmarking study, we need access to real-world datasets. However, the current dataset module only contains a collection of simple/toy forecasting datasets. To address this limitation, I suggest incorporating more complex datasets from two public repositories: 

1. [Monash Forecasting](https://forecastingdata.org/)
2. [Time Series Extrinsic Regression](http://tseregression.org/)

The goal is to create a function similar to `load_UCR_UEA_dataset` for classification tasks. This function would fetch datasets from the public repositories, download them to the local machine, and load them into memory into a format that is compatible with estimators.

Monash Forecasting has several panel datasets that the current module does not include see #3465 , one panel dataset example from the repo is  *London Smart Meters*.  The other repository is for a regression task, so it should also address issue #4314. Personally, I think it would be better to have a function for users to download the datasets, as this would reduce the memory footprint of the package and offer many more datasets for users to choose from.

This feature would not only benefit the benchmarking framework, but also users who are interested in working with real-world time series examples. With just a single line of code to download and load the dataset, users can focus on modeling rather than data collection side of things.

**Comment by hazrulakmal** [comment]
I believe that those functionalities like `make_time_series_dataset` is not  **real-world** datasets, but rather synthetic ones generated based on some characteristics. Therefore, I suggest creating a separate issue for this matter.

**Comment by fkiraly** [comment]
> If I'm not mistaken, there is a separate issue for this matter.

yes, afaik it´s this one: https://github.com/sktime/sktime/issues/4098
